### PR TITLE
[UI] [Louie] Transaction Order UI

### PR DESCRIPTION
**Pull request by cezro**
Clickup Subtask: https://app.clickup.com/t/865cput4y

- Order UI buttons and listing
- Pending and Order Request contrast not implemented

https://github.com/ezerssss/unihub/assets/90561528/494322c8-8309-41b9-a735-aa7af13dfd76

### CODE REVIEW

File: screens/Transactions/Transactions.tsx [hunk 1]
```diff
@@ -111,13 +120,13 @@ export default function Transactions({ navigation }: RootNavigationProps) {
           <Text className={`mb-1 text-xl font-normal ${titleStyle}`}>
             {product.title}
           </Text>
-          <Text className="mb-5 text-[#3838FC]">by YOU</Text>
+          <Text className="mb-5 text-[#3838FC]">
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:123]: export color to a constant

**Review by ezerssss — CHANGES_REQUESTED** [review]
Can you

- make the rounded buttons smaller, and the font size smaller (screen seems to be too tight)
- have a fixed 1/2 height for both your orders and your listings
- if no items sa that status maybe put an indicator or like text na no items or smth (refer: https://reactnative.dev/docs/flatlist#listemptycomponent)

**Comment by cezro** [comment]
> * if no items sa that status maybe put an indicator or like text na no items or smth (refer: https://reactnative.dev/docs/flatlist#listemptycomponent)

i think theres something already like that, however doesn't show up tho

might be an issue with the `.length` method something like if it exists, tried replacing it with `.length !== 0` but still doesn't work. maybe missed something here

```ts
  const renderNoOrders = !isLoading && !orders.length && (
    <Text className="mx-3 text-sm text-gray-400">
      You do not have any orders.
    </Text>
  );
  const renderNoListings = !isLoading && !listings.length && (
    <Text className="mx-3 h-14 text-sm text-gray-400">
      Your products have no orders.
    </Text>
  );
  ```

File: screens/Transactions/RoundedButton.tsx [hunk 2]
```diff
@@ -11,11 +11,13 @@ export default function RoundedButton({
...
 }: RoundedButtonProps) {
-  const textClasses = `text-white ${isActive ? '' : 'text-primary-500'}`;
+  const textClasses = `text-white text-[8px] ${
+    isActive ? '' : 'text-primary-500'
+  }`;
 
   return (
     <TouchableOpacity
-      className={`mx-1 flex min-w-[130px] flex-row justify-center rounded-full border border-primary-300 px-5 py-5 align-middle ${
+      className={`max- mx-1 flex min-w-[85px] flex-row items-center justify-center rounded-full border border-primary-300 align-middle ${
```
Comment (ezerssss) [inline screens/Transactions/RoundedButton.tsx:20]: `max-` remove

File: screens/Transactions/Transactions.tsx [hunk 3]
```diff
@@ -98,7 +147,8 @@ export default function Transactions({ navigation }: RootNavigationProps) {
 
     const titleStyle = !isSeen && 'text-primary-100';
 
-    const renderChat = activeListingStatus !== StatusEnum.CANCEL &&
+    const renderChatListing = activeListingStatus !== StatusEnum.PENDING &&
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:150]: dont remove the `StatusEnum.CANCEL` na logic

File: screens/Transactions/Transactions.tsx [hunk 4]
```diff
@@ -107,9 +157,20 @@ export default function Transactions({ navigation }: RootNavigationProps) {
           <ChatIcon />
         </TouchableOpacity>
       );
-
-    const renderEditButton = seller === user?.displayName && (
-      <TouchableOpacity className="absolute bottom-5 right-5 rounded-full bg-[#FFD700] p-[10px]">
+    const renderChatOrder = user?.displayName !== seller &&
+      activeListingStatus !== StatusEnum.DENY && (
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:161]: also add for `!== StatusEnum.CANCEL`

File: screens/Transactions/Transactions.tsx [hunk 5]
```diff
@@ -218,15 +291,28 @@ export default function Transactions({ navigation }: RootNavigationProps) {
...
+          data={
+            activeListingStatus !== StatusEnum.PENDING
+              ? listings.filter(
+                  (listing) => listing.status === activeListingStatus
+                )
+              : products.filter((product) => {
+                  return !listings.some((listing) => {
+                    return listing.product.title === product.product.title;
+                  });
+                })
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:313]: export to a constant

File: screens/Transactions/Transactions.tsx [hunk 6]
```diff
@@ -52,13 +53,58 @@ export default function Transactions({ navigation }: RootNavigationProps) {
...
+        querySnapshot.forEach((doc) => {
+          const product: Product = doc.data() as Product;
+          const tempTransaction = new TempTransaction(product);
+
+          productsArray.push(tempTransaction);
+        });
+
+        setProducts(productsArray);
+      }
+    );
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:107]: you can just store the data as a product, so the state can just be a `Product` state

File: screens/Transactions/Transactions.tsx (hunk 3 above)
Comment (ezerssss) [reply screens/Transactions/Transactions.tsx:150]:
```
const isNotCancelledOrDenied = activeListingStatus !== StatusEnum.CANCEL && activeListingStatus !== StatusEnum.DENY;

const renderChatListing = activeListingStatus !== StatusEnum.PENDING && isNotCancelledOrDenied && user?.displayName === seller
```

**Review by ezerssss — APPROVED** [review]

**Pull request merged**